### PR TITLE
Introduce CodeQL security vulnerability analysis

### PR DESCRIPTION
**Pull request by Stephan202**
Suggested commit message:
```
Introduce CodeQL security vulnerability analysis (#573)

See https://codeql.github.com
```

See the commit history for some things I tried and tested. I temporarily introduced code that referenced MD5 to trigger a warning. That code was flagged at the relevant source code line (see collapsed message). [Here](https://github.com/PicnicSupermarket/error-prone-support/security/code-scanning?query=is%3Aclosed+pr%3A573) one can see the resolved violation.

### CODE REVIEW

**Review by oxkitsune — APPROVED** [review]
Super neat stuff :rocket:

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Analyzes the code base using Github's default CodeQL query database.
```
Comment (rickie) [inline .github/workflows/codeql.yml]:
```suggestion
# Analyzes the code base using GitHub's default CodeQL query database.
```

**Review by rickie — APPROVED** [review]
Added a tiny commit. Nice commit history, really useful to go through. 

Agree that using a cron is nice for this :).

**Comment by Stephan202** [comment]
Tnx for the review!

**Pull request merged**